doc/implicit_conversion.rdoc

```diff
@@ -0,0 +1,187 @@
+== Implicit Conversion
```

Comment:
Do you think it would be nicer to call this "Implicit Conversions" (plural form).

Additionally, while this documentation focuses on implicit conversions, I wonder if we don't also talk about other concepts, and maybe a more general title might be applicable. That being said, maybe having specific (shorter) documents is better than more general (longer) documents.

According to @mame, the naming convention for the `Kernel::Array`, `Kernel::Integer` should be called "Type Conversion Functions" (functions because they are not using state like methods).

So, I wonder then if we shouldn't call this document "Type Conversions" or "Type Conversion Functions".

I don't want to shave the yak too much, so can you just make a decision? We may end up revisiting it in the future anyway.